pkg/liqo-controller-manager/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -615,11 +541,21 @@ func (r *ForeignClusterReconciler) checkTEP(ctx context.Context,
 	return nil
 }
 
-func isTunnelEndpointReason(reason string) bool {
-	switch reason {
-	case tunnelEndpointNotFoundReason, tunnelEndpointAvailableReason, tunnelEndpointConnectingReason, tunnelEndpointErrorReason:
-		return true
-	default:
-		return false
+func (r *ForeignClusterReconciler) foreignclusterEnqueuer(obj client.Object) []ctrl.Request {
+	gvks, _, err := r.Scheme.ObjectKinds(obj)
+	// Should never happen, but if it happens we panic.
+	utilruntime.Must(err)
+
+	klog.V(4).Infof("handling resource %q of type %q", klog.KObj(obj), gvks[0].String())
```

Comment:
This risks to panic in case no GVKs are found (although it should never happen IIUC).